server/pulp/server/webservices/views/roles.py

```diff
@@ -0,0 +1,223 @@
+                kwargs={'role_id': role['id']})}
+        role.update(link)
+        return generate_json_response_with_pulp_encoder(role)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, role_id):
+        """
+        Delete a  role.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param role_id: id for the requested role
+        :type role_id: str
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.role_manager()
+        result = manager.delete_role(role_id)
+        return generate_json_response(result)
+
+    @auth_required(authorization.UPDATE)
+    def put(self, request, role_id):
+        """
+        Update a specific role.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param role_id: id for the requested role
+        :type role_id: str
+        :return: Response containing the role
+        :rtype: django.http.HttpResponse
+        """
+        role_data = request.body_as_json
+        delta = role_data.get('delta', None)
+        manager = factory.role_manager()
+        role = manager.update_role(role_id, delta)
+        link = {"_href": reverse('role_resource',
+                kwargs={'role_id': role['id']})}
+        role.update(link)
+        return generate_json_response_with_pulp_encoder(role)
+
+
+class RoleUsersView(View):
+    """
+    Views for user membership within a role
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, role_id):
+        """
+        List Users belonging to a role.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param role_id: id for the requested role
+        :type role_id: str
+        :return: Response containing the users
+        :rtype: django.http.HttpResponse
+        """
+        user_query_manager = factory.user_query_manager()
+        role_users = user_query_manager.find_users_belonging_to_role(role_id)
+        return generate_json_response_with_pulp_encoder(role_users)
+
+    @auth_required(authorization.UPDATE)
+    def post(self, request, role_id):
+        """
+        Add user to a role.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param role_id: id for the requested role
+        :type role_id: str
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        :raises: InvalidValue some parameters are invalid
+        """
+        params = request.body_as_json
+        login = params.get('login', None)
+        if login is None:
+            raise pulp_exceptions.InvalidValue(login)
+
+        role_manager = factory.role_manager()
+        add_user = role_manager.add_user_to_role(role_id, login)
+        return generate_json_response(add_user)
```

Comment:
https://github.com/pulp/pulp/blob/master/server/pulp/server/managers/auth/role/cud.py#L242

The role manager appears not to return anything, so I think this would be better as:

    role_manager.add_user_to_role(role_id, login)
    return generate_json_response(None)